draft-ietf-taps-interface.md

```diff
@@ -1257,11 +1257,14 @@ Messages for streaming large data is also supported (see {{send-partial}}).
 Messages are sent on a Connection using the Send action:
 
 ~~~
-Connection.Send(messageData, messageContext?, endOfMessage?)
+messageContext := Connection.Send(messageData, messageContext?, endOfMessage?)
```

Comment:
What is this messageContext returned by Send()? If you pass in a messageContext, is it the same one?

Reply:
Yes – you only need it if you don't provide one. 
This is necessary to get rid of the msgRef

Reply:
This is to allow replying, correct? I'd say that if the application needed to get a reply, it should use an explicit messageContext instead.

My concern with returning a value that most often is not needed is that you then are often ignoring a return value, which is a warning often.

Reply:
Okay … I see your point.

This design was not made for replying, but to allow to match errors to sends without creating an explicit message context first. I have no strong opinion what is worse - having a return value that is often ignored or needing to create a message context for matching error events to sends - so let's make this an issue and decide in Montreal whether to drop the return value.

For replying, one would use the ```MessageContext``` returned by the ```recv``` event for this. Also for matching replies, using  I would suspect rather using an explicit message context than saving the implicitly generated one.